rust-runtime/aws-smithy-types/src/date_time/de.rs

```diff
@@ -0,0 +1,146 @@
+/*
+ * Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+use super::*;
+use serde::de::{Error, Visitor};
+use serde::Deserialize;
+
+struct DateTimeVisitor;
+
+struct NonHumanReadableDateTimeVisitor;
+
+impl<'de> Visitor<'de> for DateTimeVisitor {
+    type Value = DateTime;
+    fn expecting(&self, formatter: &mut fmt::Formatter<'_>) -> fmt::Result {
+        formatter.write_str("expected RFC-3339 Date Time")
+    }
+
+    fn visit_str<E>(self, v: &str) -> Result<Self::Value, E>
+    where
+        E: serde::de::Error,
+    {
+        match DateTime::from_str(v, Format::DateTime) {
+            Ok(e) => Ok(e),
+            Err(e) => Err(Error::custom(e)),
+        }
+    }
+}
+
+impl<'de> Visitor<'de> for NonHumanReadableDateTimeVisitor {
+    type Value = DateTime;
+    fn expecting(&self, formatter: &mut fmt::Formatter<'_>) -> fmt::Result {
+        formatter.write_str(
+        r#"DateTime type expects a tuple of i64 and u32 when deserializing from non human readable format.  
+        ```rust
+            //  Example
+            NonHumanReadable::Value::TupleWith2Values((
+                NonHumanReadable::Value::Int64(2354235i64),
+                NonHumanReadable::Value::UInt32(1000u32),
+            ));
+        ```"#
+        )
+    }
+
+    fn visit_seq<A>(self, mut seq: A) -> Result<Self::Value, A::Error>
+    where
+        A: serde::de::SeqAccess<'de>,
+    {
+        match seq.size_hint() {
+            Some(2) | None => match (seq.next_element()?, seq.next_element()?) {
+                (Some(seconds), Some(subsecond_nanos)) => Ok(DateTime {
+                    seconds,
+                    subsecond_nanos,
+                }),
+                _ => return Err(Error::custom("datatype mismatch")),
+            },
+            _ => Err(Error::custom("Size mismatch")),
+        }
+    }
+}
+
+impl<'de> Deserialize<'de> for DateTime {
+    fn deserialize<D>(deserializer: D) -> Result<Self, D::Error>
+    where
+        D: serde::Deserializer<'de>,
+    {
+        if deserializer.is_human_readable() {
+            deserializer.deserialize_str(DateTimeVisitor)
+        } else {
+            deserializer.deserialize_tuple(2, NonHumanReadableDateTimeVisitor)
+        }
+    }
+}
+
+#[test]
```

Comment:
```suggestion
#[cfg(test)]
```

Reply:
oops, sorry I pushed before checking. It's fixed now